pageserver/src/walreceiver.rs

```diff
@@ -469,12 +469,9 @@ async fn timeline_wal_broker_loop_step(
             updates = broker_subscription.fetch_data() => match updates {
                 Some(mut all_timeline_updates) => {
                     if let Some(subscribed_timeline_updates) = all_timeline_updates.remove(&id) {
-                        match wal_connection_manager.select_connection_candidate(subscribed_timeline_updates) {
-                            Some(candidate) => {
-                                info!("Switching to different safekeeper {} for timeline {id}, reason: {:?}", candidate.safekeeper_id, candidate.reason);
-                                wal_connection_manager.change_connection(candidate.safekeeper_id, candidate.wal_producer_connstr).await;
-                            },
-                            None => {}
+                        if let Some(candidate) = wal_connection_manager.select_connection_candidate(subscribed_timeline_updates) {
```

Comment:
is this accidental?

Reply:
No, it's because Clippy in 1.58 is somewhat stricter than in 1.60, see https://github.com/rust-lang/rust-clippy/issues/8928